face/infra/bio-sdk-resolver/src/main/java/com/simprints/face/infra/biosdkresolver/FaceBioSDK.kt

```diff
@@ -3,10 +3,15 @@ package com.simprints.face.infra.biosdkresolver
 import com.simprints.face.infra.basebiosdk.detection.FaceDetector
 import com.simprints.face.infra.basebiosdk.initialization.FaceBioSdkInitializer
 import com.simprints.face.infra.basebiosdk.matching.FaceMatcher
+import com.simprints.face.infra.basebiosdk.matching.FaceSample
 
 interface FaceBioSDK {
     val initializer: FaceBioSdkInitializer
     val detector: FaceDetector
-    val matcher: FaceMatcher
+
     val version: String
+    val templateFormat: String
+    val matcherName: String
+
+    fun createMatcher(samples: List<FaceSample>): FaceMatcher
```

Comment:
Can we name these `probeSamples` as it's unclear what samples these are until you look at the implementation?